ndc-client/src/models.rs

```diff
@@ -350,13 +350,46 @@ pub enum Aggregate {
 }
 // ANCHOR_END: Aggregate
 
+// ANCHOR: NestedObject
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[schemars(title = "NestedObject")]
+pub struct NestedObject {
+    pub fields: IndexMap<String, Field>,
```

Comment:
Sorry I just noticed this - this seems to imply that we accept `Field::Relationship` here, i.e. relationships from nested fields. Was that intended? @abhinav-hasura Is that compatible with what you're thinking on the engine side?

Also, `Field::Column` is confusing, because presumably it would not refer to a _column_, but the name of a property in the nested object?

I'm thinking we should have `NestedField` always denote a projection function (only removing or renaming things), and that `NestedObject` should always be forced to choose from the fields of the "current object".

Reply:
I'm not sure why we wouldn't want to allow relationships from nested fields. Is there some sort of problem that creates? I have a [draft PR for the engine](https://github.com/hasura/v3-engine/pull/255) which I've been working in conjunction with this spec PR and haven't run into any issues there so far.

Reply:
So I think we need to be clear about how `Field`s are meant to be evaluated in the context of a nested object. Is there a "current object" just like we have a current row? So `Column` points to the column in the current object now? 

Also, `Relationship` might be useful, but there's no way to inform the client that a field on a nested object is a good candidate for a relationship (like `foreign_keys`).

We'll pick this up in our architecture sync today and I'll drop notes here.